From 1fd13300224c7dfc467e7107e9ac7970dc855519 Mon Sep 17 00:00:00 2001
From: Auke Kok <auke-jan.h.kok@intel.com>
Date: Thu, 7 Feb 2019 21:49:22 -0800
Subject: [PATCH] Support building without SSLv3

---
 socket.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/socket.c b/socket.c
index e090bb6..cd5e84c 100644
--- a/socket.c
+++ b/socket.c
@@ -904,10 +904,13 @@ int SSLOpen(int sock, char *mycert, char *mykey, const char *myproto, int certck
 	_ssl_context[sock] = NULL;
 	int avoid_v3 = SSL_OP_NO_SSLv3;
 	if (myproto) {
+#ifndef OPENSSL_NO_SSL3_METHOD
 		if(!strcasecmp("ssl3",myproto) || !strcasecmp("ssl3.0",myproto) || !strcasecmp("sslv3",myproto) || !strcasecmp("sslv3.0",myproto)) {
 		    _ctx[sock] = SSL_CTX_new(SSLv3_client_method());
 		    avoid_v3 &= ~SSL_OP_NO_SSLv3;
-		} else if(!strcasecmp("tls1.2",myproto) || !strcasecmp("tlsv1.2",myproto)) {
+		} else
+#endif
+		if(!strcasecmp("tls1.2",myproto) || !strcasecmp("tlsv1.2",myproto)) {
 		    _ctx[sock] = SSL_CTX_new(TLSv1_2_client_method());
 		} else if(!strcasecmp("tls1.1",myproto) || !strcasecmp("tlsv1.1",myproto)) {
 		    _ctx[sock] = SSL_CTX_new(TLSv1_1_client_method());
-- 
2.20.1

